### PR TITLE
[YUNIKORN-2651]Update the unchecked error for make lint warnings

### PR DESCRIPTION
**Pull request by rich7420**
### What is this PR for?
As title, fix error check message in lint.

### What type of PR is it?
 - Bug Fix
 - Improvement
### Todos
- other lint error message.

### What is the Jira issue?
- [YUNIKORN-2651](https://issues.apache.org/jira/browse/YUNIKORN-2651)

### How should this be tested?
reomve `--new-from-rev` in Makefile 
then run 'make lint'

### Questions:
N/A

### CODE REVIEW

File: pkg/scheduler/objects/application_state.go [hunk 1]
```diff
@@ -129,97 +129,149 @@ func eventDesc() fsm.Events {
 			Src:  []string{Completed.String(), Failed.String(), Rejected.String()},
 			Dst:  Expired.String(),
 		},
-	}
-}
+		fsm.Callbacks{
+			// The state machine is tightly tied to the Application object.
```
Comment (chia7712) [inline pkg/scheduler/objects/application_state.go:133]: please revert unrelated changes ...

**Comment by rich7420** [comment]
@chia7712 thank you for your feedback. I modified them.

File: pkg/events/event_system_test.go [hunk 2]
```diff
@@ -49,7 +49,11 @@ func TestSimpleStartAndStop(t *testing.T) {
 // should be retrieved from the EventStore
 func TestSingleEventStoredCorrectly(t *testing.T) {
 	Init()
-	eventSystem := GetEventSystem().(*EventSystemImpl) //nolint:errcheck
+	eventSystem, ok := GetEventSystem().(*EventSystemImpl)
+	if !ok {
+		t.Fatalf("Failed to cast GetEventSystem() to *EventSystemImpl")
+		return
```
Comment (chia7712) [inline pkg/events/event_system_test.go:55]: we don't need this `return`, right?

File: pkg/events/event_system_test.go [hunk 3]
```diff
@@ -82,7 +86,11 @@ func TestSingleEventStoredCorrectly(t *testing.T) {
 
 func TestGetEvents(t *testing.T) {
 	Init()
-	eventSystem := GetEventSystem().(*EventSystemImpl) //nolint:errcheck
+	eventSystem, ok := GetEventSystem().(*EventSystemImpl)
+	if !ok {
+		t.Fatalf("Failed to cast GetEventSystem() to *EventSystemImpl")
+		return
```
Comment (chia7712) [inline pkg/events/event_system_test.go:92]: ditto

File: pkg/events/event_system_test.go [hunk 4]
```diff
@@ -114,7 +122,11 @@ func TestConfigUpdate(t *testing.T) {
 	defer configs.SetConfigMap(map[string]string{})
 
 	Init()
-	eventSystem := GetEventSystem().(*EventSystemImpl) //nolint:errcheck
+	eventSystem, ok := GetEventSystem().(*EventSystemImpl)
+	if !ok {
+		t.Fatalf("Failed to cast GetEventSystem() to *EventSystemImpl")
+		return
```
Comment (chia7712) [inline pkg/events/event_system_test.go:128]: ditto

File: pkg/log/logger_test.go [hunk 5]
```diff
@@ -243,7 +243,9 @@ func TestParentLogger(t *testing.T) {
 
 func resetTestLogger() {
 	// flush log
-	logger.Sync() //nolint:errcheck
+	if err := logger.Sync(); err != nil {
+		t.Fatalf("Error syncing logger: %s", err.Error())
```
Comment (chia7712) [inline pkg/log/logger_test.go:247]: please use `assert.NilError`

File: pkg/scheduler/objects/application_state.go [hunk 6]
```diff
@@ -140,7 +140,11 @@ func eventDesc() fsm.Events {
 func callbacks() fsm.Callbacks {
 	return fsm.Callbacks{
 		"enter_state": func(_ context.Context, event *fsm.Event) {
-			app := event.Args[0].(*Application) //nolint:errcheck
+			app, ok := event.Args[0].(*Application)
```
Comment (chia7712) [inline pkg/scheduler/objects/application_state.go:143]: This will increase complexity. Maybe we can keep origin `nolint`

File: pkg/scheduler/partition_test.go [hunk 7]
```diff
@@ -3817,7 +3817,11 @@ func TestTryAllocateMaxRunning(t *testing.T) {
 
 func TestNewQueueEvents(t *testing.T) {
 	events.Init()
-	eventSystem := events.GetEventSystem().(*events.EventSystemImpl) //nolint:errcheck
+	eventSystem, ok := events.GetEventSystem().(*events.EventSystemImpl)
+	if !ok {
+		t.Fatalf("Failed to cast GetEventSystem() to *EventSystemImpl")
+		return
```
Comment (chia7712) [inline pkg/scheduler/partition_test.go:3823]: ditto

File: pkg/scheduler/objects/queue_test.go [hunk 8]
```diff
@@ -2455,7 +2455,11 @@ func TestQueue_setAllocatingAccepted(t *testing.T) {
 
 func TestQueueEvents(t *testing.T) {
 	events.Init()
-	eventSystem := events.GetEventSystem().(*events.EventSystemImpl) //nolint:errcheck
+	eventSystem, ok := events.GetEventSystem().(*events.EventSystemImpl)
+	if !ok {
+		t.Fatalf("Failed to cast GetEventSystem() to *EventSystemImpl")
+		return
```
Comment (chia7712) [inline pkg/scheduler/objects/queue_test.go:2461]: ditto

File: pkg/scheduler/objects/application_test.go [hunk 9]
```diff
@@ -2198,7 +2214,11 @@ func TestPlaceholderLargerEvent(t *testing.T) {
 	app := newApplication(appID1, "default", "root.default")
 	// Create event system after new application to avoid new application event.
 	events.Init()
-	eventSystem := events.GetEventSystem().(*events.EventSystemImpl) //nolint:errcheck
+	eventSystem, ok := events.GetEventSystem().(*events.EventSystemImpl)
+	if !ok {
+		t.Fatalf("Failed to cast GetEventSystem() to *EventSystemImpl")
+		return
```
Comment (chia7712) [inline pkg/scheduler/objects/application_test.go:2220]: ditto

File: pkg/scheduler/objects/application_test.go [hunk 10]
```diff
@@ -2079,7 +2091,11 @@ func TestAllocationEvents(t *testing.T) { //nolint:funlen
 	app := newApplication(appID1, "default", "root.default")
 	// Create event system after new application to avoid new app event.
 	events.Init()
-	eventSystem := events.GetEventSystem().(*events.EventSystemImpl) //nolint:errcheck
+	eventSystem, ok := events.GetEventSystem().(*events.EventSystemImpl)
+	if !ok {
+		t.Fatalf("Failed to cast GetEventSystem() to *EventSystemImpl")
+		return
```
Comment (chia7712) [inline pkg/scheduler/objects/application_test.go:2097]: ditto

File: pkg/scheduler/objects/application_test.go [hunk 11]
```diff
@@ -2013,7 +2021,11 @@ func TestAskEvents(t *testing.T) {
...
```
Comment (chia7712) [inline pkg/scheduler/objects/application_test.go:2027]: ditto

File: pkg/scheduler/objects/application_test.go [hunk 12]
```diff
@@ -380,7 +384,11 @@ func TestAddAllocAsk(t *testing.T) {
 	app := newApplication(appID1, "default", "root.unknown")
 	// Create event system after new application to avoid new application event.
 	events.Init()
-	eventSystem := events.GetEventSystem().(*events.EventSystemImpl) //nolint:errcheck
+	eventSystem, ok := events.GetEventSystem().(*events.EventSystemImpl)
+	if !ok {
+		t.Fatalf("Failed to cast GetEventSystem() to *EventSystemImpl")
+		return
```
Comment (chia7712) [inline pkg/scheduler/objects/application_test.go:390]: ditto

**Review by chia7712 — COMMENTED** [review]
@rich7420 please check other unchecked error in code base

**Comment by chia7712** [comment]
@rich7420 I feel you misunderstand the issue. The purpose of YUNIKORN-2651 is to handle the remaining unchecked error warned by lint. You can see the reports by `make lint` (you have to remove `--new-from-rev`). If there is no unchecked error warnings, please feel free to close this PR
